animation/animation.ino

```diff
@@ -0,0 +1,394 @@
+/*
+This is the code for the animation of the tree lights. It is
+responsible for controlling lights and setting the animations etc.
+*/
+
+#include <FastLED.h>
+
+///////////////////////////////////////
+// Hardware definitions
+///////////////////////////////////////
+
+#define PIN_LED 13
+#define NUM_LEDS 178
+
+
+#define DATA_PIN 5
+#define BUTTON_PIN 6
+
+///////////////////////////////////////
+// Timing Variables
+///////////////////////////////////////
+
+uint32_t last_draw = 0;
+uint32_t wait_draw = 10;
+
+uint32_t last_update = 0;
+uint32_t wait_update = 10; // This is the number of MS between frames. e.g. 500 = 2 fps
+
+uint32_t last_buttoncheck = 0;
+uint32_t wait_buttoncheck = 10;
+
+uint32_t flashing_start_time = 0;
+
+// Function prototypes for the animations
+void rainbow_vibe();
+void vibe_cursor_spin();
+void vibe_pulse();
+void vibe_sparkle();
+
+///////////////////////////////////////
+// Light Variables and Animation Variables
+///////////////////////////////////////
+
+bool is_button_pressed = false; //Check for button press
+bool last_buttonpress = false; // This variable stores whether the button was pressed in the last loop
+
+uint8_t global_hue = 0; //Sets hue of CHSV to 0
+uint8_t pos = 0; //initial location of the cursor
+uint8_t bri = 255; //brightness of LEDs
+int dir = -1; //direction of brightness change for pulse animation
+
+int chance_of_firework = 3; //set the likelihood of a firework going off in the firework animation
+bool go_firework = false; //variable to store whether firework should be shot off or not
+
+//Define the array of LEDs
+CRGB leds[NUM_LEDS];
+
+//Store animations in array
+void (*animationFunctions[])() = {rainbow_vibe, vibe_cursor_spin, vibe_pulse, vibe_sparkle};
+
+int numAnimations = sizeof(animationFunctions) / sizeof(animationFunctions[0]);
+
+
+//Setup steps go here
+void setup()
+{
+  
+  Serial.begin(115200);
+  delay(1000); // Put a small delay to ensure everything is setup properly
+  FastLED.addLeds<NEOPIXEL, DATA_PIN>(leds, NUM_LEDS);
+  pinMode(PIN_LED, OUTPUT);
+  digitalWrite(PIN_LED, HIGH); // Turn the built-in led on to show there is power
+  pinMode(BUTTON_PIN, INPUT_PULLUP);
+  fill_solid(leds, NUM_LEDS, 0x000000); //turn all leds off
+  FastLED.show();
+  
+  FastLED.setBrightness(100); //Set Brightness of LED strip
```

Comment:
Great seeing this here to reduce wattage while testing. Once moved to final hardware make sure to bump the brightness up